compiler/x/codegen/OMRTreeEvaluator.cpp

```diff
@@ -4587,16 +4587,180 @@ OMR::X86::TreeEvaluator::vfmaEvaluator(TR::Node *node, TR::CodeGenerator *cg)
    return resultReg;
    }
 
+TR::Register* OMR::X86::TreeEvaluator::SIMDreductionEvaluator(TR::Node* node, TR::CodeGenerator* cg)
+   {
+   TR::Node *child = node->getFirstChild();
+   TR::Register *inputVectorReg = cg->evaluate(child);
+   TR::VectorLength vl = child->getDataType().getVectorLength();
+   TR::DataType dt = child->getDataType().getVectorElementType();
+   TR::VectorOperation op = node->getOpCode().getVectorOperation();
+   bool useGPR = !dt.isFloatingPoint();
+
+   TR::InstOpCode regOpcode = getNativeSIMDOpcode(OMR::ILOpCode::reductionToVerticalOpcode(node->getOpCodeValue(), vl), child->getDataType(), false);
+
+   TR::Register *workingReg = cg->allocateRegister(TR_VRF);
+   TR::Register *resultVRF = cg->allocateRegister(TR_VRF);
+   TR::Register *fprReg = dt.isFloatingPoint() ? cg->allocateRegister(TR_FPR) : NULL;
+   TR::Register *tmpNaNReg = dt.isFloatingPoint() ? cg->allocateRegister(TR_VRF) : NULL;
+   TR::Register *rSrcReg;
+
+   TR::LabelSymbol *reduce64Label = generateLabelSymbol(cg);
+   TR::LabelSymbol *check2 = generateLabelSymbol(cg);
+   TR::LabelSymbol *doneLabel = generateLabelSymbol(cg);
+   reduce64Label->setStartInternalControlFlow();
+   doneLabel->setEndInternalControlFlow();
```

Comment:
These two labels seem to be used only to create an internal control flow region, but is the ICF region needed for anything? I don't see the reason for it

Reply:
The original plan was to cmp and jmp if one of the results was NaN. I forgot to delete this.